MangaRipper.Core/Services/Downloader.cs

```diff
@@ -51,23 +54,60 @@ private async Task<string> DownloadStringAsyncInternal(string url, CancellationT
             var request = CreateRequest();
             using (var response = await request.GetAsync(url, cancellationToken))
             {
+                GetAllCloudFlareCookies(response);
                 return await response.Content.ReadAsStringAsync();
             }
         }
 
         private HttpClient CreateRequest()
         {
+            var cookieContainer = new CookieContainer();
+            // Set CloudFlare cookies if we have one
+            if (CacheProvider.Instance.CacheExists("__cfduid"))
+            {
+                cookieContainer.Add(CacheProvider.Instance.GetCacheValue("__cfduid"));
```

Comment:
The unit test shows that GetCacheValue can return null and throw an exception. Please check null before add cookie.

https://ci.appveyor.com/project/NguyenDanPhuong/mangaripper/build/1.0.348

Reply:
Damn, you right. Forgot to add the "cf_clearance" to the condition.